internal/server/receiver_handlers.go

```diff
@@ -155,9 +231,15 @@ func (s *ReceiverServer) validate(ctx context.Context, receiver apiv1.Receiver,
 		if err != nil {
 			return fmt.Errorf("unable to validate HMAC signature: %s", err)
 		}
+		r.Body = io.NopCloser(bytes.NewReader(b))
 		return nil
 	case apiv1.GitHubReceiver:
-		_, err := github.ValidatePayload(r, []byte(token))
+		b, err := io.ReadAll(r.Body)
```

Comment:
The logic of loading the body into memory and recreating a reader for it is duplicated in a few places. Can we extract it on its own `func`?

This process would benefit by adopting a `io.LimitReader` to limit the amount of data being loaded into memory to a reasonable/configurable amount.